### PR TITLE
feat: add levels to `Trace_core`.

### PR DESCRIPTION
**Pull request by c-cube**
these levels are used to control the verbosity levels.

### CODE REVIEW

File: src/core/level.ml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    level [Debug1] is enabled, everything below it (Error, Warning, Info, etc.)
+    are also enabled.
+    @since NEXT_RELEASE *)
+type t =
+  | Error  (** Only errors *)
+  | Warning  (** Warnings *)
+  | Info
+  | Debug1  (** Least verbose debugging level *)
+  | Debug2  (** Intermediate verbosity debugging level *)
+  | Debug3  (** Maximum verbosity debugging level *)
```
Comment (tatchi) [inline src/core/level.ml:21]: What's the reason for having multiple `Debug` levels? Isn't one enough?

**Review by tatchi — COMMENTED** [review]
Should we add tests? I'm getting an error when I set the current log level to `Info` in `t1.ml`

```diff
git diff
diff --git a/test/t1.ml b/test/t1.ml
index 87e4792..a4c6e32 100644
--- a/test/t1.ml
+++ b/test/t1.ml
@@ -1,6 +1,7 @@
 let run () =
   Trace.set_process_name "main";
   Trace.set_thread_name "t1";
+  Trace.set_current_level Trace.Level.Info;
 
```

```sh
File "test/dune", line 3, characters 8-10:
3 |   (name t1)
            ^^
trace-tef error: cannot find span -9223372036854775808
trace-tef error: cannot find span -92
```

Looks like it's trying to exit a `span` that wasn't created due to `check_level` being `false`

https://github.com/c-cube/ocaml-trace/blob/e81650defb9a6a3afb1a472e7da4b454baa99488/src/core/trace_core.ml#L40-L47

**Comment by tatchi** [comment]
> Looks like it's trying to exit a span that wasn't created due to check_level being false

I wonder how it will behave with `message`. Say we have a current log level equals to `Info` and we do:

```ocaml
Trace.with_span ~level:Trace.Level.Trace "main" @@ fun span ->
  Trace.message ~level:Trace.Level.Info ~span "world";
```

I'm not sure exactly how `message` is supposed to work but looking at this code I understand that we want to "attach" a message to a `span`. But since the span was created with a level > `Info`, it maybe won't exist?

**Comment by c-cube** [comment]
Thanks a lot for finding the bug in t1.ml btw, I just fixed it.

**Pull request merged**